src-js/fontra-webcomponents/src/custom-data-list.js

```diff
@@ -0,0 +1,347 @@
+      model.customData = newCustomData;
+      addRemoveButton.scrollIntoView({
+        behavior: "auto",
+        block: "nearest",
+        inline: "nearest",
+      });
+      labelList.setSelectedItemIndex(items.length - 1);
+    };
+
+    labelList.addEventListener("deleteKey", deleteSelectedItem);
+    const addRemoveButton = html.createDomElement("add-remove-buttons", {
+      addButtonCallback: async () => {
+        const currentKeys = labelList.items.map((customData) => {
+          return customData.key;
+        });
+        const { key, value } = await this._customDataPropertiesRunDialog(
+          this.customDataInfos,
+          currentKeys
+        );
+        if (key == undefined || value == undefined) {
+          return;
+        }
+        const newItem = makeItem([key, value]);
+        const newItems = [...labelList.items, newItem];
+        model.customData = newItems.map((label) => {
+          return { ...label };
+        });
+        labelList.setItems(newItems);
+        labelList.editCell(newItems.length - 1, "key");
+        addRemoveButton.scrollIntoView({
+          behavior: "auto",
+          block: "nearest",
+          inline: "nearest",
+        });
+      },
+      removeButtonCallback: deleteSelectedItem,
+      disableRemoveButton: true,
+    });
+
+    updateRemoveButton(labelList, addRemoveButton);
+
+    return html.div({ style: "display: grid; grid-gap: 0.3em;" }, [
+      labelList,
+      addRemoveButton,
+    ]);
+  }
+
+  render() {
+    this.contentElement.appendChild(this.buildCustomDataList());
+  }
+
+  async _customDataPropertiesRunDialog(customDataInfos, currentKeys) {
+    const title = translate("Add advanced information"); // TODO: translation
+    const customDataNames = this.customDataInfos.map((info) => info.key);
+
+    const validateInput = () => {
+      warningElement.innerText = "";
+      infoElement.innerText = "";
+      const customDataKey =
+        nameController.model.customDataKey == ""
+          ? undefined
+          : nameController.model.customDataKey;
+      const customDataValue =
+        nameController.model.customDataValue == ""
+          ? undefined
+          : nameController.model.customDataValue;
+
+      const setWarning = (warning) => {
+        warningElement.innerText = warning;
+        dialog.defaultButton.classList.toggle("disabled", true);
+      };
+
+      if (customDataKey == undefined && customDataValue == undefined) {
+        // We don't want to start with a warning,
+        // but need to disable the button -> therefore set it to empty string.
+        setWarning("");
+        return;
+      }
+
+      if (customDataKey == undefined) {
+        setWarning("⚠️ Key is empty, please enter."); // TODO: translation
+        return;
+      }
+
+      if (!customDataNames.includes(customDataKey)) {
+        // We know we have a key, but it is not supported, yet.
+        setWarning(`⚠️ ${translate("Key not supported, yet.")}`); // TODO: translation
```

Comment:
The "yet" makes no sense for garbage input, better to remove it.

Reply:
Thanks. Done.